common/include/pcl/impl/point_types.hpp

```diff
@@ -645,6 +645,28 @@ namespace pcl
       a = 255;
     }
 
+    inline PointXYZRGB(float _x, float _y, float _z)
+    {
+      x = _x;
+      y = _y;
+      z = _z;
+      data[3] = 1.0f;
+      r = g = b = 0;
+      a = 255;
+    }
+
+    inline PointXYZRGB(float _x, float _y, float _z, std::uint8_t _r, std::uint8_t _g, std::uint8_t _b)
```

Comment:
You can achieve the same by defaulting the r,g,b values to 0 and removing the other ctor

Reply:
I would not call it the same. As it is now, the user either specifies each of R, G, and B values, or none. With your proposal, it's possible to write something like

```cpp
PointXYZRGB pt(0, 0, 0, 128)
```

which I believe is rather confusing for an unprepared reader.

Reply:
I see. But isn't `PointXYZRGB(10, 10, 10, 20, 20, 20)` just as confusing?

Reply:
Perhaps I'm biased, but no, not so confusing. The type name spells X-Y-Z-R-G-B, so a constructor with 6 arguments feel unambiguous.

Reply:
OK. Let's go with the original commit then.